src/lib/merkle_ledger/database.ml

```diff
@@ -414,4 +398,28 @@ module Make
   let merkle_path_at_index_exn t index =
     let addr = Addr.of_int_exn index in
     merkle_path_at_addr_exn t addr
+
+  let addr_to_location addr = Location.Account addr
+end
+
+module Make
+    (Key : Intf.Key)
+    (Account : Intf.Account with type key := Key.t)
+    (Hash : Intf.Hash with type account := Account.t)
+    (Depth : Intf.Depth)
+    (Location : Location_intf.S)
+    (Kvdb : Intf.Key_value_database)
+    (Storage_locations : Intf.Storage_locations) :
+  Database_intf.S
+  with module Location = Location
+   and module Addr = Location.Addr
+   and type account := Account.t
+   and type root_hash := Hash.t
+   and type hash := Hash.t
+   and type key := Key.t
+   and type key_set := Key.Set.t = struct
+  module Base =
+    Make0 (Key) (Account) (Hash) (Depth) (Location) (Kvdb) (Storage_locations)
```

Comment:
After seeing `Make0`, I don't like the indirection and I think it makes things a bit harder to read. 

So instead, with the original `Make` functor, include the `Util.Make` like the following:


```ocaml
module Make .... = struct
.....

include Util.Make(Location)(Account) (Location.Addr) (struct 
    type nonrec t = t
    let get = get
    let  end)


end

```

I don't expect `Base` to be a module with a lot of inputs in the future, so it's okay to write the `Base` input like how I did in the above example.